core/chains/evm/txmgr/eth_broadcaster.go

```diff
@@ -533,134 +541,48 @@ func (eb *EthBroadcaster) handleInProgressEthTx(ctx context.Context, etx EthTx,
-		// any transaction that gets insufficient_eth. This is OK if a
-		// transaction with a large VALUE blocks because this always comes last
-		// in the processing list.
-		// If it blocks because of a transaction that is expensive due to large
-		// gas limit, we could have smaller transactions "above" it that could
-		// theoretically be sent, but will instead be blocked.
-		return sendError, true
-	}
-
-	if sendError == nil {
-		// We want to observe the time until the first _successful_ broadcast.
-		// Since we can re-enter this method by way of tryAgainBumpingGas,
-		// and we pass the same initialBroadcastAt timestamp there, when we re-enter
-		// this function we'll be using the same initialBroadcastAt.
 		observeTimeUntilBroadcast(eb.chainID, etx.CreatedAt, time.Now())
 		return eb.orm.UpdateEthTxAttemptInProgressToBroadcast(&etx, attempt, EthTxAttemptBroadcast, func(tx pg.Queryer) error {
 			return eb.incrementNextNonceAtomic(tx, etx)
 		}), true
-	}
-
-	// In the case of timeout, we fall back to the backoff retry loop and
-	// subsequent tries ought to resend the exact same in-progress transaction
-	// attempt and get a definitive answer on what happened
-	if sendError.IsTimeout() {
-		return errors.Wrapf(sendError, "timeout while sending transaction %s (eth_tx ID %d)", attempt.Hash.Hex(), etx.ID), true
-	}
-
-	// Unknown error here. All bets are off in this case, it is possible the
-	// transaction could have been accepted. We may be running on an
-	// unsupported RPC or chain.
-	//
-	// The most conservative course of action would be to retry this
-	// transaction forever (or until success) however this can lead to nodes
-	// getting stuck if we are on an unsupported new chain.
-	//
-	// We can continue in a kind of gracefully degraded manner if we check the
-	// chain for its view on our latest nonce. If it has been incremented, then
-	// it accepted the transaction despite the error and we can move forwards
-	// assuming success in this case.
-
-	if sendError.IsTxFeeExceedsCap() {
-		lgr.Criticalw(fmt.Sprintf("Sending transaction failed; %s", label.RPCTxFeeCapConfiguredIncorrectlyWarning),
-			"etx", etx,
-			"attempt", attempt,
-			"err", sendError,
-			"id", "RPCTxFeeCapExceeded",
-		)
-		eb.SvcErrBuffer.Append(sendError)
-		// Note that we may have broadcast to multiple nodes and had it
-		// accepted by one of them! It is not guaranteed that all nodes share
-		// the same tx fee cap. That is why we must treat this as an unknown
-		// error that may have been confirmed.
+	case txmgrtypes.Underpriced:
+		return eb.tryAgainBumpingGas(ctx, lgr, err, etx, attempt, initialBroadcastAt)
+	case txmgrtypes.InsufficientFunds:
+		fallthrough
+	case txmgrtypes.Retryable:
+		return err, true
+	case txmgrtypes.Unsupported:
+		if eb.config.ChainType().IsL2() {
+			return eb.tryAgainWithNewEstimation(ctx, lgr, err, etx, attempt, initialBroadcastAt)
+		}
+		return err, false
+	default:
```

Comment:
nit: can we change to txmtypes.Unknown? Should exhaust all cases right

Reply:
+1

Reply:
Also, seems like same logic is used for error type ExceedsFeeCap too.
So do this:

case txmgrtypes.ExceedsFeeCap:
                 // This means primary RPC node rejected this tx. However other SendOnlyNodes could've accepted it, and this Tx may confirm. So treat this as another case of Unknown error.
                 lgr.Criticalw('RPC node rejected this tx as outside Fee Cap')
		fallthrough
case txmgrtypes.Unknown:
.
.
.